sktime/datatypes/_panel/_check.py

```diff
@@ -211,7 +211,8 @@ def check_pdmultiindex_panel(obj, return_metadata=False, var_name="obj", panel=T
 
     # check that no dtype is object
     if "object" in obj.dtypes.values:
-        msg = f"{var_name} should not have column of 'object' dtype"
+        obj_cols = obj.select_dtypes(include=["object"]).columns
+        msg = f"{var_name} should not have column of 'object' dtype. Column:{obj_cols}"
```

Comment:
this could be clearer, I would say: `"should not have columns of "object" dtype, but found columns: {obj_cols}"`